### ISSUE TITLE
QTable, mask=True and FalseArray

### ISSUE THREAD

**Issue by abeelen** [astropy/astropy#16495]
### Description

The behavior of the `masked=True` is counter intuitive. If used, it assign an `astropy.columns.FalseArray` to each column, which in principle can not be set to `True` but  an inplace change could set the value to True silently, although it is not used later.

The inplace changes is fundamental when using very long table as it avoids memory allocation.

### Expected behavior

The inplace operation should fail as per the `FalseArray` definition or, I believe a better solution, use a proper numpy array for the mask in `QTable`, which is actually the case for `Table`

```python 
from astropy.table import Table
tab = Table([[1, 1]], names=['test'], masked=True)
#tab['test'].mask is array([False, False])
```

### How to Reproduce

```python 
import astropy.units as u
from astropy.table import QTable
tab = QTable([[1, 1] * u.mJy], names=['test'], masked=True)
# tab['test'].mask is FalseArray([False, False])
# tab['test'].mask[0] = True # Will raise a `ValueError` as expected from `FalseArray` 
tab['test'].mask |= [True, True] # Will work without a warning
# tab['test'] is now  FalseArray([ True,  True])
```

### Versions

Linux-6.7.9-amd64-x86_64-with-glibc2.38
Python 3.11.9 (main, Apr 10 2024, 13:16:36) [GCC 13.2.0]
astropy 6.0.1
Numpy 1.26.4
pyerfa 2.0.1.4
Scipy 1.11.4
Matplotlib 3.6.3

**Comment by mhvk** [comment]
Hmm, I guess the simplest solution would be to make `FalseArray` read-only, but it would be nicer to just use `MaskedQuantity`, just like `MaskedColumn` is used for `Table`. I actually have an old branch that meant to do that, but I never pursued it far enough so it is without test cases, etc. Maybe time to revive it... https://github.com/astropy/astropy/compare/main...mhvk:astropy:table-allow-conversion-masked-column-to-masked-quantity?expand=1

**Comment by mhvk** [comment]
A fix is in #16500, though it still needs more tests and a changelog entry.

**Comment by mhvk** [comment]
Hmm, this is fixed by #16500 not #16498, sorry!
